packages/open-api-gateway/src/project/open-api-gateway-ts-project.ts

```diff
@@ -0,0 +1,243 @@
+    this.package.addField(
+      "main",
+      path.join(this.libdir, this.srcdir, "index.js")
+    );
+    this.package.addField(
+      "types",
+      path.join(this.libdir, this.srcdir, "index.d.ts")
+    );
+
+    // Set to private since this either uses workspaces or has file dependencies
+    this.package.addField("private", true);
+
+    // Generated project should have a dependency on this project, in order to run the generation scripts
+    this.addDeps(OPENAPI_GATEWAY_PDK_PACKAGE_NAME, "constructs", "aws-cdk-lib");
+
+    // Synthesize the openapi spec early since it's used by the generated typescript client, which is also synth'd early
+    const spec = new OpenApiSpecProject({
+      name: `${this.name}-spec`,
+      parent: this,
+      outdir: this.specDir,
+    });
+    spec.synth();
+
+    const codegenDir = "generated";
+
+    new TextFile(this, path.join(codegenDir, "README.md"), {
+      lines: [
+        "## Generated Clients",
+        "",
+        "This directory contains generated client code based on your OpenAPI Specification file (spec.yaml).",
+        "",
+        "Like other `projen` managed files, this directory should be checked in to source control, but should not be edited manually.",
+      ],
+      readonly: true,
+    });
+
+    // Parent the generated code with this project's parent for better integration with monorepos
+    this.hasParent = !!(options.parent && options.outdir);
+    const codegenDirRelativeToParent = this.hasParent
+      ? path.join(options.outdir!, codegenDir)
+      : codegenDir;
+
+    // We generate the typescript client since this project will take a dependency on it
+    this.generatedTypescriptClient = new GeneratedTypescriptClientProject({
+      parent: this.hasParent ? options.parent : this,
+      defaultReleaseBranch: options.defaultReleaseBranch,
+      name: `${this.package.packageName}-typescript`,
+      outdir: path.join(codegenDirRelativeToParent, ClientLanguage.TYPESCRIPT),
+      // Use the parsed spec such that refs are resolved to support multi-file specs
+      specPath: spec.parsedSpecPath,
+      packageManager: options.packageManager,
+    });
+
+    // Synth early so that the generated code is available prior to this project's install phase
+    this.generatedTypescriptClient.synth();
+
+    const typescriptCodeGenDir = path.relative(
+      this.outdir,
+      this.generatedTypescriptClient.outdir
+    );
+
+    if (this.hasParent) {
+      // When we have a parent project, we can add a dependency on the generated typescript client since it's part of
+      // the monorepo. Since this project will be synthesized before the parent monorepo, the generated typescript
+      // client won't be visible for the first install in this project's post synthesize step, so we use a local
+      // workspace as a workaround.
+      if (this.package.packageManager === NodePackageManager.PNPM) {
+        new YamlFile(this, "pnpm-workspace.yaml", {
```

Comment:
We should probably expose this is an optional public property incase the user ever needs to modify it.